.ci/watchdog/SlackCommunicator.py

```diff
@@ -41,13 +41,13 @@ class SlackCommunicator:
         :type slack_token:        str
     """
 
-    def __init__(self, slack_token):
-        self.thread_id = None
+    def __init__(self, slack_token, proxy):
```

Comment:
You can avoid adding another parameter here if you detect the proxy inside the function.
You can always override with:
```
slack_app.proxy = custom_proxy
```

Reply:
I like that idea. In last commit I changed approach to setting default proxy from env variables and also exposing it as property.